src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +323,66 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
+				// just in `inputNetwork` before merging it to `shaderNetwork` ... but this would require
+				// dealing with weird connections where the input node handle is relative to `shaderNetwork`,
+				// but the output handle is relative to `inputNetwork`. This gets very confusing if there are
+				// nodes in the two networks with the same name, which get uniquified during addShaders.
+				// Doing this after merging simplifies all that.
+				std::vector<IECore::InternedString> shadersToDelete;
+				for( const auto &i : shaderNetwork->shaders() )
+				{
+					if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
```

Comment:
We're nested pretty deep here already. Feels like a good candidate for `if( !ShaderTweakProxy::isProxy() ) { continue; }` so that the meat of the loop can lose a level of indentation. Or maybe this could be pushed out into a separate function?

Reply:
Did the continue thing - missed the suggestion about separate function.